contracts/cryptoblades.sol

```diff
@@ -271,6 +271,49 @@ contract CryptoBlades is Initializable, AccessControlUpgradeable {
         require(foundMatch, "Target invalid");
     }
 
+    mapping (address => uint32) tickets;
+
+    function spendTicket(uint32 num)
+        public
+    {
+        require (tickets[msg.sender] >= num, "You don't have enough tickets");
+
+        for (int i = 0; i < num; i++) {
+            weapons.mint(msg.sender, randoms.getRandomSeed(msg.sender));
+        }
+
+        tickets[msg.sender] -= num;
+    }
+    
+    function isUnlikely(uint24 pp, uint24 ep)
+        private
+        pure
+        returns(bool)
+    {
+        int128 playerMin = ABDKMath64x64.fromUInt(pp).mul(ABDKMath64x64.fromUInt(90)).div(ABDKMath64x64.fromUInt(100));
+        int128 playerMax = ABDKMath64x64.fromUInt(pp).mul(ABDKMath64x64.fromUInt(110)).div(ABDKMath64x64.fromUInt(100));
+        int128 playerRange = playerMax.sub(playerMin);
+        int128 enemyMin = ABDKMath64x64.fromUInt(ep).mul(ABDKMath64x64.fromUInt(90)).div(ABDKMath64x64.fromUInt(100));
+        int128 enemyMax = ABDKMath64x64.fromUInt(ep).mul(ABDKMath64x64.fromUInt(110)).div(ABDKMath64x64.fromUInt(100));
+        int128 enemyRange = enemyMax.sub(enemyMin);
+        int256 rollingTotal = 0;
+        
+        if (playerMin >= enemyMin) {
+            int128 temp = playerMin.sub(enemyMin).div(enemyRange);
+            temp = temp.add(ABDKMath64x64.fromUInt(1).sub(temp).mul(playerMax.sub(enemyMax).div(playerRange)));
+            temp = temp.add(ABDKMath64x64.fromUInt(1).sub(temp).mul(ABDKMath64x64.fromUInt(50).div(ABDKMath64x64.fromUInt(100))));
+            rollingTotal = ABDKMath64x64.toInt(temp.mul(ABDKMath64x64.fromUInt(1000)));
+        } else {
+            int128 temp = enemyMin.sub(playerMin).div(playerRange);
+            temp = temp.add(ABDKMath64x64.fromUInt(1).sub(temp).mul(enemyMax.sub(playerMax).div(enemyRange)));
+            temp = temp.add(ABDKMath64x64.fromUInt(1).sub(temp).mul(ABDKMath64x64.fromUInt(50).div(ABDKMath64x64.fromUInt(100))));
+            temp = ABDKMath64x64.fromUInt(1).sub(temp);
+            rollingTotal = ABDKMath64x64.toInt(temp.mul(ABDKMath64x64.fromUInt(1000)));
+        }
+
+        return rollingTotal <= 300 ? true : false;
```

Comment:
This logic probably does not require use of `ABDKMath64x64` to implement at all, and removing all uses of it should lead to huge gas savings.

Reply:
It may be possible to convert the calculations, but there won't be big gas benefits. I did a quick test in remix and the gas changed from 28277 to 25490.